### DISCUSSION TITLE
"ReferenceError: useOutsideClick is not defined" in website

### DISCUSSION THREAD

**Discussion by keemtj** [chakra-ui/chakra-ui#8862] [General]
### Description

When I entered Chakra-UI web page, an error had already occurred. check the error please

**Chakra UI Version**
2.8.2

### Problem Statement/Justification

Problem: Reference error occur in the example code of the library documentation, making it difficult for users to follow the examples.

Proposal: It is expected that resolving error in the example code can address user confusion when reading the documentation.

Reasons:
Improved user convenience: Users can receive support to understand and resolve issues with example code.
Enhanced documentation quality: Enhancing the quality of library documentation can provide users with a better experience.

### Proposed Solution or API

-

### Additional Information

please check this link and screenshots

[https://v2.chakra-ui.com/docs/hooks/use-outside-click](https://v2.chakra-ui.com/docs/hooks/use-outside-click)

![스크린샷 2024-04-19 오전 10 13 43](https://github.com/chakra-ui/chakra-ui/assets/51189962/d586f762-5cb0-412f-8c85-1cd0a1bafcf2)

**Comment by Saloni-hub** [comment]
I am checking github repo they are not added any custom hook useOutSideClick u can make your hook and enable above properties like this

import React, { useRef, useState, useEffect } from "react";
import { ChakraProvider, Box } from "@chakra-ui/react";
import ReactDOM from "react-dom";

function useOutsideClick(ref, handler) {
useEffect(() => {
function handleClickOutside(event) {
handler(event);
}

document.addEventListener("mousedown", handleClickOutside);
return () => {
  document.removeEventListener("mousedown", handleClickOutside);
};
}, [ref, handler]);
}

function Example() {
const [isModalOpen, setIsModalOpen] = useState(false);
const modalRef = useRef(null);

useOutsideClick(modalRef, () => {
setIsModalOpen(false);
});

return (
<>
<button onClick={() => setIsModalOpen(true)}>Open Modal
{isModalOpen && (
<div
ref={modalRef}
style={{
position: "fixed",
top: 0,
left: 0,
width: "100%",
height: "100%",
background: "rgba(0, 0, 0, 0.5)",
}}
>
<div
style={{
position: "absolute",
top: "50%",
left: "50%",
transform: "translate(-50%, -50%)",
background: "white",
padding: "20px",
borderRadius: "5px",
}}
>
👋 Hey, I'm a modal. Click anywhere outside of me to close.


)}
</>
);
}

function App() {
return (



);
}

const rootElement = document.getElementById("root");
ReactDOM.render(
<React.StrictMode>



</React.StrictMode>,
rootElement
);

**Comment by jpenna** [comment]
There are other pages showing this `ReferenceError` banner:

`useDimensions`
https://v2.chakra-ui.com/docs/hooks/use-dimensions#usage
![image](https://github.com/user-attachments/assets/cc6c178c-8071-42d4-a21c-d618d6ee8688)

`useConst`
https://v2.chakra-ui.com/docs/hooks/use-const#usage
![image](https://github.com/user-attachments/assets/2b08e23d-27eb-4050-ab85-03a653b5643b)

**Comment by segunadebayo** [comment]
`useOutsideClick` was removed. Use `useClickAway` from `react-use`.
